File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        ans = []
        for c in s:
            ans.append(c)
            if len(ans) >=3 and ans[-3] == ans [-2] == ans[-1]:
                ans.pop()
        return "".join(ans) 

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.js
/**
 * @param {string} s
 * @return {string}
 */
var makeFancyString = function(s) {
   let ans = [];
    for (let i = 0; i < s.length; i++) {
        ans.push(s[i]);
        let n = ans.length;
        if (n >= 3 && ans[n - 1] === ans[n - 2] && ans[n - 2] === ans[n - 3]) {
            ans.pop();
        }
    }
    return ans.join("");
};